### PR TITLE
Aded support for oauth and mtls.

### PR DESCRIPTION
**Pull request by nmaqsood91**
This PR contains the following functionalities 

- enable Oauth security scheme
- enable the key choice for Oauth

- [x] Updated unit tests (`*.spec.js`)  
- [x] Updated e2e tests ([here](https://github.com/ramgrandhi/serverless-wso2-apim/tree/main/src/__tests__/e2e))  
- [ ] Updated documentation ([here](https://github.com/ramgrandhi/serverless-wso2-apim/blob/main/README.md))   

🙌 Thank you!

### CODE REVIEW

File: src/3.2.0/wso2apim.js [hunk 1]
```diff
@@ -200,10 +200,12 @@ async function constructAPIDef(user, gatewayEnv, apiDef, apiId) {
       securityScheme.push('mutualssl');
       securityScheme.push('mutualssl_mandatory');
     }
-    if(apiDef.securityScheme && apiDef.securityScheme.oauth2 && apiDef.securityScheme.oauth2.enabled === false) {
-      //do nothing
-    } else {
-      securityScheme.push('oauth2');
+    if (apiDef.securityScheme && apiDef.securityScheme.oauth2 && apiDef.securityScheme.oauth2.enabled === true) {
```
Comment (diego-antonelli) [inline src/3.2.0/wso2apim.js:203]: This condition in now inversed from what it used to be, before it was ignoring this case but now line 206 was moved to inside this if. Is this the intended behavior?

File: src/__tests__/e2e/valid-mtls-enabled-and-oauth-enabled-3.2.0/serverless.yml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  name: aws
+  stackName: ${env:STACK_NAME}
+  deploymentBucket:
+    name: ${env:TEST_ID_NORMALIZED}
+plugins:
+  - serverless-localstack
+  - serverless-deployment-bucket
+  - "../../../../../src"
+
+#⬇⬇⬇⬇⬇⬇⬇⬇⬇⬇⬇⬇⬇⬇ Modify the configuration below to suit your test case.
```
Comment (diego-antonelli) [inline src/__tests__/e2e/valid-mtls-enabled-and-oauth-enabled-3.2.0/serverless.yml:12]: Can you remove unnecessary comments?

**Review by diego-antonelli — CHANGES_REQUESTED** [review]

File: src/3.2.0/wso2apim.js (hunk 1 above)
Comment (nmaqsood91) [reply src/3.2.0/wso2apim.js:203]: Yes I inverted it and it should not be like this what it was before

File: src/__tests__/e2e/valid-mtls-enabled-and-oauth-enabled-3.2.0/serverless.yml (hunk 2 above)
Comment (nmaqsood91) [reply src/__tests__/e2e/valid-mtls-enabled-and-oauth-enabled-3.2.0/serverless.yml:12]: that is in all e2e test cases, for the sake of keeping things same I did not touch it. Do you want me to remove it?

**Review by diego-antonelli — APPROVED** [review]

**Pull request merged**